src/coreclr/vm/assembly.hpp

```diff
@@ -557,6 +558,12 @@ class Assembly
     void CacheManifestFiles();
 
     void CacheFriendAssemblyInfo();
+#ifndef DACCESS_COMPILE
+    ReleaseHolder<FriendAssemblyDescriptor> GetFriendAssemblyInfo(bool *pfNoFriendAssemblies);
```

Comment:
Are the copy constructors, etc. on ReleaseHolder setup correctly to make it safe to use as return value?

Reply:
(We typically stay away from returning Holder types.)

Reply:
At some point the ReleaseHolder type I'm using was updated to have the appropriate move constructor to make this work safely. I've verified that with the new test case, the type is freed at the appropriate moment, and not earler, or later. In some work I've done where I could use tools like unique_ptr, I have found the pattern of allowing such returns to be a nice upgrade to readability.